tests/pytests/test_util.py

```diff
@@ -30,9 +76,72 @@ def test_kernel_from_cube_order(cube_kernels):
     assert kernels == ['targetposition0', 'targetposition1','instrumentposition', 'instrumentpointing0', 'instrumentpointing1', 'attitudeshape', 'instrument', 'clock']
 
 def test_kernel_from_cube_no_kernel_group():
-   with pytest.raises(KeyError):
-       with tempfile.NamedTemporaryFile('r+') as cube:
+    with pytest.raises(KeyError):
+       with tempfile.NamedTemporaryFile('w+') as cube:
            cube.write('')
            cube.flush()
            util.generate_kernels_from_cube(cube.name)
 
+def test_get_preferences_arg(tmpdir, pvl_one_group):
+    with open(tmpdir.join('IsisPrefrences'), 'w+') as pvl_file:
+        pvl_file.write(pvl_one_group)
+        pvl_file.flush()
+
+        pvl_obj = util.get_isis_preferences(pvl_file.name)
+        pvl_obj_from_dict = util.get_isis_preferences({**pvl_obj})
+
+        assert pvl_obj['Test']['t'] == 't1'
+        assert pvl_obj == pvl_obj_from_dict
+
+def test_get_prefernces_arg_isisroot(monkeypatch, tmpdir, pvl_one_group, pvl_two_group):
+    monkeypatch.setenv('ISISROOT', str(tmpdir))
+
+    with open(tmpdir.join('IsisPreferences'), 'w+') as pvl_isisroot_file:
+        with open(tmpdir.join('arg_prefs.pvl'), 'w+') as pvl_arg_file:
+            pvl_arg_file.write(pvl_two_group)
+            pvl_arg_file.flush()
+
+            pvl_isisroot_file.write(pvl_one_group)
+            pvl_isisroot_file.flush()
+
+            pvl_obj = util.get_isis_preferences(pvl_arg_file.name)
+
+            assert pvl_obj['Test']['t'] == 't2'
+            assert pvl_obj['Data']['b'] == 'b2'
+            assert pvl_obj['Data']['a'] == 'a2'
+            assert pvl_obj['Data']['r'] == 'r2'
+
+def test_dict_to_lower():
+    data = {'F': {'O' : {'O': 1}, 'B': {'A': 2}}}
+    expected  = {'f': {'o' : {'o': 1}, 'b': {'a': 2}}}
+    assert util.dict_to_lower(data) == expected
+
+def test_get_prefernces_arg_isisroot_home(monkeypatch, tmpdir, pvl_one_group, pvl_two_group, pvl_three_group):
+    tmpdir.mkdir('.Isis')
+    monkeypatch.setenv('ISISROOT', str(tmpdir))
+    monkeypatch.setenv('HOME', str(tmpdir))
```

Comment:
Do these go away after the test finishes?

Reply:
yes